### DISCUSSION TITLE
How to identify AEAD-enabled messages using PGPainless

### DISCUSSION THREAD

**Discussion by msfjarvis** [pgpainless/pgpainless#440] [Q&A] [answered]
Over at Password Store we've had a steady stream of users coming in with AEAD enabled keys and running into problems, which I've added detection for during key import but I'd also like to warn users when a password entry was created with an AEAD enabled key. Any help with building in such a detection mechanism would be greatly appreciated :)

**Comment by vanitasvitae** [comment, marked as answer]
The AEAD feature is denoted by the `Features` subpacket.
You could probably search the latest direct-key signature and primary user-id binding signature for the features subpacket and check, if `SignatureSubpacketsUtil.parseFeatures(signature).contains(Feature.MODIFICATION_DETECTION_2)`

Edit: I will soon work on adding OpenPGP v6 features to BC, so stay tuned for native support :)

**Reply by msfjarvis** [reply]
Great to hear! I'm assuming the GnuPG shenanigans are going to stay unsupported?

**Reply by vanitasvitae** [reply]
I'll assure basic read-support, so that GnuPG artifacts can be consumed, but my priorities are v6.
